jvm-packages/xgboost4j-gpu/src/main/java/ml/dmlc/xgboost4j/gpu/java/ColumnDMatrix.java

```diff
@@ -0,0 +1,96 @@
+/*
+ Copyright (c) 2021 by Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+package ml.dmlc.xgboost4j.gpu.java;
+
+import java.util.Iterator;
+
+import ml.dmlc.xgboost4j.java.DMatrix;
+import ml.dmlc.xgboost4j.java.XGBoostError;
+
+/**
+ * DMatrix based on GPU
+ */
+class ColumnDMatrix extends DMatrix {
```

Comment:
Why is it called colum matrix?  The comment in constructor uses `DeviceQuantileDMatrix`.

Reply:
DMatrix created by columnars. Any suggestions?

Reply:
The underlying implementation is not limited to columnar matrix, it can be arbitrary data type.  Let's not restrict ourselves to cudf.

Reply:
any suggestion for the new name? Note: This class may create the Normal DMatrix or DeviceDMatrix